### PR TITLE
Add rosco package

### PR DESCRIPTION
**Pull request by marchdf**
- Move the rosco package from a local repo to spack
- removal from exawind-manager and discussion is done here: https://github.com/Exawind/exawind-manager/pull/42)
- added patches for intel oneapi
- ~~Only maintain 2.9.1 and above since there was a cmake reorg that changed things around.~~
- ~~2.9.0 works with intel oneapi, other versions fail~~

### CODE REVIEW

**Comment by marchdf** [comment]
@spackbot style

**Review by psakievich — CHANGES_REQUESTED** [review]
If we know this will break with oneapi for speficic version we should embed that in package logic.

See the first example in this link: https://spack.readthedocs.io/en/latest/packaging_guide.html#conflicts-and-requirements

**Comment by marchdf** [comment]
@psakievich I've added the patches for intel oneapi and tested them on kestrel.

**Review by psakievich — APPROVED** [review]

**Comment by ndevelder** [comment]
I think this looks good. Seems like "CMAKE_Fortran_FLAGS" is interchangeable between OneAPI and Intel? Or at least the flags are non breaking? 

I wasn't intending for Dan and I to be the official maintainers when I put this in the local repo, but I'm fine with staying on. We may want to reach out to Dan Z. to see how he feels about it? Anyone else we think would want to be a package maintainer for this?

File: var/spack/repos/builtin/packages/rosco/package.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ROSCO controls package for OpenFAST from NREL
+    Note: this only builds ROSCO controls library for inclusion with OpenFAST
+    If the toolbox or tuning scripts are needed, please build manually
+    """
+
+    homepage = "https://rosco.readthedocs.io/en/latest/"
+    url = "https://github.com/NREL/ROSCO/archive/refs/tags/v2.9.0.tar.gz"
+    git = "https://github.com/NREL/ROSCO.git"
+
+    maintainers("dzalkind", "ndevelder")
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/rosco/package.py:20]: We need to confirm willingness to be maintainers.  Sounds like @ndevelder is in.  @dzalkind please let @marchdf know if you want to be listed as a maintainer for this package.

**Review by psakievich — CHANGES_REQUESTED** [review]

**Comment by marchdf** [comment]
> Seems like "CMAKE_Fortran_FLAGS" is interchangeable between OneAPI and Intel? Or at least the flags are non breaking?

Yes they are recognized by oneapi.

File: var/spack/repos/builtin/packages/rosco/package.py (hunk 1 above)
Comment (dzalkind) [reply var/spack/repos/builtin/packages/rosco/package.py:20]: Sure, I am willing.

**Review by psakievich — DISMISSED** [review]

File: var/spack/repos/builtin/packages/rosco/package.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rosco/package.py]: Not required this year but GTH when adding a new package.

```suggestion
# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — COMMENTED** [review]
FYI Confirmed homepage, branches, and all version sha256s.

**Review by psakievich — APPROVED** [review]

**Pull request merged**